### PR TITLE
planner: remove task argument from IndexHashJoin.GetCost and IndexMergeJoin.GetCost

### LINKED ISSUE #33934: planner: remove `task` argument from `IndexHashJoin.GetCost` and `IndexMergeJoin.GetCost`
## Enhancement
No logical change, just remote the `task` argument from `IndexHashJoin.GetCost` and `IndexMergeJoin.GetCost` to let them keep uniform with other operators.

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #33934

Problem Summary: planner: remove task argument from IndexHashJoin.GetCost and IndexMergeJoin.GetCost

### What is changed and how it works?

No logical change, just remote the `task` argument from `IndexHashJoin.GetCost` and `IndexMergeJoin.GetCost` to let them keep uniform with other operators.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -5304,6 +5304,23 @@ func TestIndexJoinCost(t *testing.T) {
 		`  ├─Selection_9(Build) 1.25 0.00 cop[tikv]  not(isnull(test.t_inner_idx.a))`,
 		`  │ └─IndexRangeScan_7 1.25 0.00 cop[tikv] table:t_inner_idx, index:a(a) range: decided by [eq(test.t_inner_idx.a, test.t_outer.a)], keep order:false, stats:pseudo`,
 		`  └─TableRowIDScan_8(Probe) 1.25 0.00 cop[tikv] table:t_inner_idx keep order:false, stats:pseudo`))
+
+	tk.MustQuery("explain format=verbose select /*+ inl_hash_join(t_outer, t_inner_idx) */ t_inner_idx.a from t_outer, t_inner_idx where t_outer.a=t_inner_idx.a").Check(testkit.Rows(
+		`IndexHashJoin_12 12487.50 221872.19 root  inner join, inner:IndexReader_9, outer key:test.t_outer.a, inner key:test.t_inner_idx.a, equal cond:eq(test.t_outer.a, test.t_inner_idx.a)`,
```
Comment (qw4990) [inline planner/core/integration_test.go:5309]: Has the same cost as Master Branch.

**Review by Reminiscent — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]

**Comment by purelind** [comment]
### TiDB MergeCI notify
**🟠 Bad News! New failing [1] after this pr merged.**
These new failed integration tests seem to be caused by the current PR, please try to fix these new failed integration tests, thanks!
CI Name | Result | Duration | Compare with Parent commit
------------ | ------------- | ------------- | -------------
[idc-jenkins-ci-tidb/common-test](https://ci.pingcap.net/job/tidb_ghpr_common_test/9472/display/redirect) | 🟥 failed 1, success 11, total 12 | 6 min 9 sec | New failing
[idc-jenkins-ci-tidb/integration-br-test](https://ci.pingcap.net/job/tidb_ghpr_integration_br_test/8785/display/redirect) | 🟥 failed 25, success 4, total 29 | 15 min | Existing failure
[idc-jenkins-ci-tidb/integration-copr-test](https://ci.pingcap.net/job/tidb_ghpr_integration_copr_test/9049/display/redirect) | ✅ all 1 tests passed | 6 min 19 sec | Fixed
[idc-jenkins-ci/integration-cdc-test](https://ci.pingcap.net/job/tidb_ghpr_integration_cdc_test/1814/display/redirect) | 🟢 all 34 tests passed | 46 min | Existing passed
[idc-jenkins-ci-tidb/code-coverage](https://ci.pingcap.net/job/tidb_ghpr_coverage/3001/display/redirect) | 🟢 Lines coverage: 63.49% | 14 min | Existing passed
[idc-jenkins-ci-tidb/tics-test](https://ci.pingcap.net/job/tidb_ghpr_tics_test/8890/display/redirect) | 🟢 all 1 tests passed | 12 min | Existing passed
[idc-jenkins-ci-tidb/integration-common-test](https://ci.pingcap.net/job/tidb_ghpr_integration_common_test/10641/display/redirect) | 🟢 all 11 tests passed | 12 min | Existing passed
[idc-jenkins-ci-tidb/sqllogic-test-2](https://ci.pingcap.net/job/tidb_ghpr_sqllogic_test_2/9136/display/redirect) | 🟢 all 28 tests passed | 9 min 27 sec | Existing passed
[idc-jenkins-ci-tidb/sqllogic-test-1](https://ci.pingcap.net/job/tidb_ghpr_sqllogic_test_1/9198/display/redirect) | 🟢 all 26 tests passed | 8 min 4 sec | Existing passed
[idc-jenkins-ci-tidb/integration-ddl-test](https://ci.pingcap.net/job/tidb_ghpr_integration_ddl_test/9261/display/redirect) | 🟢 all 6 tests passed | 7 min 12 sec | Existing passed
[idc-jenkins-ci-tidb/mybatis-test](https://ci.pingcap.net/job/tidb_ghpr_mybatis/9060/display/redirect) | 🟢 all 1 tests passed | 4 min 45 sec | Existing passed
[idc-jenkins-ci-tidb/integration-compatibility-test](https://ci.pingcap.net/job/tidb_ghpr_integration_campatibility_test/10048/display/redirect) | 🟢 all 1 tests passed | 4 min 44 sec | Existing passed
[idc-jenkins-ci-tidb/plugin-test](https://ci.pingcap.net/job/tidb_merged_pr_build/2870/display/redirect) | 🟢 build success, plugin test success | 4min | Existing passed

**Pull request merged**